src/fparser/scripts/read.py

```diff
@@ -63,36 +64,56 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
 # DAMAGE.
 
-import os
+'''Python script with command line options which calls the Fortran
+File Reader with the supplied filename(s) and outputs the readers
+representation of the code(s).
+
+'''
+from __future__ import print_function
 import sys
-### START UPDATE SYS.PATH ###
-### END UPDATE SYS.PATH ###
+
 try:
     from iocbio.optparse_gui import OptionParser
 except ImportError:
     from optparse import OptionParser
-from fparser.script_options import set_read_options
+from fparser.scripts.script_options import set_read_options
+
+
+def runner(_, options, args):
+    '''Call the Fortran File reader for each filename in args and print
+    out its content.
 
-def runner (parser, options, args):
-    from fparser.readfortran import  FortranFileReader
+    :param options: command line argument information from the options \
+    parser
+    :type options: :py:class:`optparse.Values`
+    :param args: a list of fortran filepaths
+    :type args: list of str
+
+    :raises NotImplementedError: if the task option is not set to \
+    "show".
+
+    '''
+    from fparser.common.readfortran import FortranFileReader
     for filename in args:
         reader = FortranFileReader(filename)
-        if options.task=='show':
+        if options.task == 'show':
             for item in reader:
-                print(item, file=sys.stdout)
+                print(item)
                 sys.stdout.flush()
         else:
-            raise NotImplementedError(repr(options.task))
-        
+            raise NotImplementedError(
+                "The task option '{0}' is invalid. Currently only "
+                "'show' is supported.".format(repr(options.task)))
 
-def main ():
+
+def main():
+    '''Check input options then call the runner function.'''
     parser = OptionParser()
     set_read_options(parser)
-    if hasattr(parser, 'runner'):
-        parser.runner = runner
     options, args = parser.parse_args()
     runner(parser, options, args)
     return
```

Comment:
pylint complains that this `return` is "useless" :-)

Reply:
My pylint (and an updated version) as well as pylint3 do not complain for some strange reason but your pylint is correct so I've removed it.